python-package/lightgbm/basic.py

```diff
@@ -61,6 +61,12 @@
     np.ndarray,
     pd_Series
 ]
+_LGBM_PositionType = Union[
+    List[float],
+    List[int],
```

Comment:
Since this is a new API, I think we should not support providing it as a list, and require that it be only a `numpy` array or `pandas` Series.

List objects are very inefficient for array processing. We support lists for the other types to avoid breaking existing user code, but for this new interface I don't think LightGBM should accept Python list objects.

Reply:
Done via [81b5f09](https://github.com/microsoft/LightGBM/pull/5929/commits/81b5f09c69f041f206bbf69cb897108a9cc72db4)